### PR TITLE
(Enhancement)Patient Visit Header Component.[O3-1288].

### PR DESCRIPTION
**Pull request by hadijahkyampeire**
[Ticket](https://issues.openmrs.org/browse/O3-1288)

## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

- Add the logic for `starting visit` using the new button in the visit header.
- Changes the visit header navbar color to brand 2.
- Cater for long names on a desktop view, truncate and display the full details in a tooltip.

## Screenshots

1. Desktop View when a patient has a long name.
Screenshot 2022-05-18 at 15 45 54

2. In Tablet mode, the name is displayed as it is. 
Screenshot 2022-05-18 at 15 46 31

3. Video showing the start visit button now working as expected .

https://user-images.githubusercontent.com/30952856/169079680-487f8d8f-0228-4e21-b1a5-a4cf9ccf6aa1.mov

## Related Issue
- https://issues.openmrs.org/browse/O3-1249

## Other

*None.*

### CODE REVIEW

**Review by ZacButko — APPROVED** [review]
This is a very nice improvement! Thank you Hadijah!

A small thing to notice. It looks like the patient is presumed to not be active until the session comes back telling that this patient has an active session, which in general I agree with. However here that makes the "start visit" button appear for a patient that already has a visit started, which on slower internet connection is visible for some time before the correct information is received. (see gif). Can we change the default behavior for this button to only show up after we have confirmed that they are not in an active visit? Maybe this is a task for a future ticket.

slow-load

File: packages/esm-outpatient-app/src/visit-header/work-spaces.ts
Comment (ZacButko) [reply packages/esm-outpatient-app/src/visit-header/work-spaces.ts:1]: @donaldkibet in this case I'm fine to write it twice in order to ignore circular dependency problems. Core does not require patient-management to run, nor should it. The most appropriate place for all of the patient management features remains in patient-management. This only duplicates the "start visit" functionality, but I don't expect this logic to change frequently and if it is not working in one space or the other we will quickly notice. Somewhat inelegant but a good solution IMHO

File: packages/esm-outpatient-app/src/visit-header/visit-header.scss
Comment (hadijahkyampeire) [reply packages/esm-outpatient-app/src/visit-header/visit-header.scss:57]: For this color I couldn't find any in `vars.scss` where we most of color variable and it looks to be new I have not seen it in another place in the app, is it possible for it to be added as well?

File: packages/esm-outpatient-app/src/visit-header/work-spaces.ts
Comment (hadijahkyampeire) [reply packages/esm-outpatient-app/src/visit-header/work-spaces.ts:1]: You are right @donaldkibet I used the same function from the `common-lib` because I think it is easier for now until we decide to move it to a module that is accessible by the `patient-management`

Comment (ibacher) [reply packages/esm-outpatient-app/src/visit-header/work-spaces.ts:1]: I do think we need to consider moving the "workspace" concept into esm-core, albeit for reasons that aren't specific to this PR. This does add a bit of a weird dependency on @openmrs/patient-chart-app to the outpatient-app, so we haven't entirely avoided the dependency issues (specifically the patient-chart-app needs to be loaded for the i18n to work here).

That said, being pragmatic about this, I think there are a number of things to work through in making the workspace concept more general and this probably isn't the PR to do it.

**Review by donaldkibet — APPROVED** [review]

**Comment by hadijahkyampeire** [comment]
@ZacButko regarding the start visit showing at first and the disappearing, I couldn't fix that since I can't tell when even the call is done loading, it only has a `isValidating` prop which is really confusing, Also the data starts when it is `null` and even after loading it stays `null` if the patient doesn't have an active visit, this makes it difficult to know when the null value is for `still loading` or actually `No active visit`, I am blocked on that, please advise how I can go about that.

**Comment by hadijahkyampeire** [comment]
@ZacButko I used the carbon color variables that @donaldkibet shared but I couldn't get away to handle the condition for waiting until the currentVisit finishes loading and then check if it exists, the hook was not return a loading prop which I think is the one that can help with that.

Also there is some check failing but I am not sure how to fix it 🤔 .
Please advise on the above issues.

**Comment by ibacher** [comment]
@hadijahkyampeire The `isValidating` property does indicate whether or not the visit is loading... More specifically, the visit is initially loading if `isValidating && data === null` and is `null` if `!isValidating && data === null`.

**Comment by hadijahkyampeire** [comment]
Thanks @ibacher let me try to use the two conditions and see.

**Comment by hadijahkyampeire** [comment]
Thanks @ibacher the conditions above fixed it, @ZacButko the button issue is now fixed I will merge this after CI builds.

**Pull request merged**
